pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/http/AsyncHttpConnector.java

```diff
@@ -176,39 +153,83 @@ private URI replaceWithNew(InetSocketAddress address, URI uri) {
         return URI.create(newUri);
     }
 
+    @Override
+    public Future<?> apply(ClientRequest jerseyRequest, AsyncConnectorCallback callback) {
+        final CompletableFuture<ClientResponse> resp = new CompletableFuture<>();
+
 
+        CompletableFuture.runAsync(() -> {
```

Comment:
I think we should rewrite this logic in async way. It seems that we try to wrap a sync code in an async way.